test/widgets/action_sheet_test.dart

```diff
@@ -610,4 +610,39 @@ void main() {
       check(mockSharePlus.sharedString).isNull();
     });
   });
+
+  group('MessageActionSheetCancelButton', () {
+    final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+
+    Finder cancelButtonFinder() => find.text(zulipLocalizations.dialogCancel);
+
+    void checkActionSheet(WidgetTester tester, {required bool isShown}) {
+      // TODO(i18n) skip translation for now
+      check(find.text('React with 👍').evaluate().length).equals(isShown ? 1 : 0);
+      check(find.text(zulipLocalizations.actionSheetOptionStarMessage)
+        .evaluate().length).equals(isShown ? 1 : 0);
+      check(find.text(zulipLocalizations.actionSheetOptionQuoteAndReply)
+        .evaluate().length).equals(isShown ? 1 : 0);
+      check(find.text(zulipLocalizations.actionSheetOptionMarkAsUnread)
+        .evaluate().length).equals(isShown ? 1 : 0);
+      check(find.text(zulipLocalizations.actionSheetOptionCopyMessageText)
+        .evaluate().length).equals(isShown ? 1 : 0);
+      check(find.text(zulipLocalizations.actionSheetOptionCopyMessageLink)
+        .evaluate().length).equals(isShown ? 1 : 0);
+      check(find.text(zulipLocalizations.actionSheetOptionShare)
+        .evaluate().length).equals(isShown ? 1 : 0);
```

Comment:
What about checking for the number of buttons? If more buttons are added/removed, this test will fail and will make us update the number. I think this will have a good representation of actions sheet being shown!

Reply:
> If more buttons are added/removed, this test will fail and will make us update the number.

I think that's the sort of update that is likely to feel like just busywork when we have to do it. See the paragraphs around the mention of "busywork" here:
https://zulip.readthedocs.io/en/latest/testing/philosophy.html#integration-testing-or-unit-testing

If there were a test that had the purpose of testing the list of options, that could be a useful test. But I think that purpose is basically covered by the individual options' tests. And the list varies based on a number of conditions, so really there can't usefully be just one test for the whole list.



> I think this will have a good representation of actions sheet being shown!

I think the best representation of the list of options in the action sheet is the one in the source code, in `showMessageActionSheet`. With a bit of cleanup to have a more uniform format, it looks like this:
```dart
      return Column(children: [
        if (!hasThumbsUpReactionVote)
          AddThumbsUpButton(message: message, messageListContext: context),
        StarButton(message: message, messageListContext: context),
        if (isComposeBoxOffered)
          QuoteAndReplyButton(message: message, messageListContext: context),
        if (showMarkAsUnreadButton)
          MarkAsUnreadButton(message: message, messageListContext: context, narrow: narrow),
        CopyMessageTextButton(message: message, messageListContext: context),
        CopyMessageLinkButton(message: message, messageListContext: context),
        ShareButton(message: message, messageListContext: context),
      ]);
```

which is hard to beat; really the only way to be more concise would be to reduce those argument lists passed to the constructors.

In general it's very often the case that the best representation of what's in the UI, or of how something should behave, is in the source code. Tests are valuable for other purposes (like preventing regressions, and thereby enabling making changes confidently) but IME for this purpose they're rarely better than the code under test.

Reply:
> With a bit of cleanup to have a more uniform format

Sent #902 for that.

Relatedly, in this PR it would be helpful to pull out the list of columns as a local variable in that build method. That will keep it from getting real deeply indented (which is helpful for not making the formatting cramped), and will also help make the list easy to spot by separating it visually from all the layout details.